sysutils/poac/Portfile

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cmake 1.1
+PortGroup           github 1.0
+PortGroup           boost 1.0
+PortGroup           openssl 1.0
+
+github.setup        poacpm poac 0.3.1
+categories          sysutils
+license             Apache-2
+
+maintainers         {@ken-matsui kmatsui.me:poac} \
+                    openmaintainer
+description         A package manager for C++ like Cargo for Rust
+long_description    Poac is a package manager for C++ like Cargo \
+                    for Rust. Poac can download project's dependencies \
+                    and compile a project.
+
+checksums           rmd160  ae4f2bddbf9fffc5ca95ecb902fa56b980796967 \
+                    sha256  986a5b66226be5bae0b601a4d79768c4af9ab901b05a6ee9ebb492916d6bed2e \
+                    size    118699
+
+configure.args-append \
+                    -DCMAKE_BUILD_TYPE=Release \
```

Comment:
we prefer you not to specify the build type, but have it use the default "MacPorts" (as defined in the `cmake` PG) that will allow it to use the flags that are added by MacPorts. You didn't have this in your earlier Portfile, is it needed for anything or caused a build failure earlier?

Reply:
@reneeotten 

It caused the following error:

```console
CMake Error at /opt/local/var/macports/build/_Users_ken-matsui_ports_sysutils_poac/poac/work/build/_deps/libarchive-src/CMakeLists.txt:35 (MESSAGE):
  Unknown keyword for CMAKE_BUILD_TYPE: MacPorts

  Acceptable keywords: Debug,Release,RelWithDebInfo,MinSizeRel
```

Reply:
@ken-matsui since it using `cmake 1.1` portgroup it is easier to set the `cmake.build_type` line instead [as here](https://github.com/macports/macports-ports/blob/13c447288d05e6773ebd98e151ca9d26f49effcd/math/vampire/Portfile#L30).

Reply:
Wow, cool! Thank you :)

Reply:
@ken-matsui have you tried building with GCC 11 like I do with the [vtm port](https://github.com/macports/macports-ports/blob/cf3e72f1c7c930f9584d5f56ef0050993b17142e/sysutils/vtm/Portfile#L30)? This is not the best advice, but I also wonder why the compilation crashes at the end.

Reply:
@aeiouaeiouaeiouaeiouaeiouaeiou 
Not yet, thank you for your information! I'll try it on https://github.com/ken-matsui/macports-ports/pull/1.

Reply:
I see that it did not help much, but now it shows this error:

```
   Undefined symbols for architecture x86_64:
    "__ZN5boost24scoped_static_mutex_lockC1ERNS_12static_mutexEb", referenced from:
        __ZN5boost16cpp_regex_traitsIcE16get_catalog_nameB5cxx11Ev in libpoac_core_builder_ninja.a(syntax.cc.o)
        __ZN5boost11basic_regexIcNS_12regex_traitsIcNS_16cpp_regex_traitsIcEEEEE9do_assignEPKcS7_j.isra.0 in libpoac_core_builder_ninja.a(syntax.cc.o)
    "__ZN5boost24scoped_static_mutex_lockD1Ev", referenced from:
        __ZN5boost16cpp_regex_traitsIcE16get_catalog_nameB5cxx11Ev in libpoac_core_builder_ninja.a(syntax.cc.o)
        __ZN5boost11basic_regexIcNS_12regex_traitsIcNS_16cpp_regex_traitsIcEEEEE9do_assignEPKcS7_j.isra.0 in libpoac_core_builder_ninja.a(syntax.cc.o)
        __ZN5boost11basic_regexIcNS_12regex_traitsIcNS_16cpp_regex_traitsIcEEEEE9do_assignEPKcS7_j.isra.0.cold in libpoac_core_builder_ninja.a(syntax.cc.o)
```

Reply:
Umm... This is weird. I'll do investigate it since I could reproduce this locally.